From ef1b52f21b449d9d09ddbbcb9ee0824619cade9d Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:40 -0400
Subject: [fs] nfsd: Remove nfs4_lock_state(): nfsd4_delegreturn()

Message-id: <1410898849-22126-157-git-send-email-bfields@redhat.com>
Patchwork-id: 95839
O-Subject: [Patch RHEL7 156/165] nfsd: Remove nfs4_lock_state(): nfsd4_delegreturn()
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 084d4d4549d88f7a27f455f9fe8a03fd9842f5ba
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index cde72d8..733e653 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -4855,7 +4855,6 @@ nfsd4_delegreturn(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  if ((status = fh_verify(rqstp, &cstate->current_fh, S_IFREG, 0)))
   return status;
 
- nfs4_lock_state();
  status = nfsd4_lookup_stateid(cstate, stateid, NFS4_DELEG_STID, &s, nn);
  if (status)
   goto out;
@@ -4868,8 +4867,6 @@ nfsd4_delegreturn(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
 put_stateid:
  nfs4_put_stid(&dp->dl_stid);
 out:
- nfs4_unlock_state();
-
  return status;
 }
 
-- 
1.7.1